debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SqlServerConnectorIT.java

```diff
@@ -1839,6 +1839,42 @@ private void restartInTheMiddleOfTx(boolean restartJustAfterSnapshot, boolean af
         }
     }
 
+    @Test
+    @FixFor("DBZ-2697")
+    public void shouldEmitNoEventsForSkippedUpdateAndDeleteOperations() throws Exception {
+        final Configuration config = TestHelper.defaultConfig()
+                .with(SqlServerConnectorConfig.SNAPSHOT_MODE, SnapshotMode.INITIAL)
+                .with(SqlServerConnectorConfig.SKIPPED_OPERATIONS, "u,d")
+                .build();
+
+        start(SqlServerConnector.class, config);
+        assertConnectorIsRunning();
+
+        // Wait for snapshot completion
+        consumeRecordsByTopic(1);
+
+        connection.execute("INSERT INTO tablea VALUES(201, 'insert201')");
+        connection.execute("UPDATE tablea SET cola='insert201-update' WHERE id=201");
+        connection.execute("INSERT INTO tablea VALUES(202, 'insert202')");
+        connection.execute("DELETE FROM tablea WHERE id=202");
+        connection.execute("INSERT INTO tablea VALUES(203, 'insert203')");
+
```

Comment:
```suggestion
TestHelper.waitForCdcRecord(connection, "tablea", rs -> rs.getInt("id") == 203);
```
I'd like to suggest we add the following so that the test method waits until SQL Server has propagated the record so that we don't run into any random timing issues with test failures.

Reply:
I'm going to use it. thanks